activerecord/lib/active_record/relation/batches.rb

```diff
@@ -233,6 +233,14 @@ def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore:
           ids = records.map(&:id)
           yielded_relation = where(primary_key => ids)
           yielded_relation.load_records(records)
+        elsif self == klass.unscoped.all
```

Comment:
We can probably move that scope out of the loop.

Reply:
Sorry `scope` -> `check`

Reply:
If we move it out of the loop, we need to have a condition like `range_batching_possible = !load && self == klass.unscoped.all` and use this variable then. We need `!load`, because otherwise the comparison will load records for association relations https://github.com/rails/rails/blob/7669a4a9757bff49bfc37d772886f59a90dd0434/activerecord/lib/active_record/association_relation.rb#L14-L16

Leaving it as is seems just cleaner, imo?

Reply:
Hum, I'm just a bit annoyed to have to instantiate two relations on every loop. Maybe we just cache `klass.unscoped.all`?

That `==` implementation is scary too. I fear we might trigger some awful query by accident, but I guess we have some tests that would fail, right?